promises-> manage multiple execution of aasynch. events 

it checks wheather event is occured or not

<script>
    const pobj1 = new Promise( (resolve, reject) => {
        let roll_no = [1,2,3,4,5];
        resolve(roll_no);
       // reject('Error while communicating');
        

    }, 2000);

        const getBiodata = () =>{
            return new Promise( (resolve, reject) => {
                setTimeout((indexdata) => {
                    let biodata = {
                        name: 'vinod',
                        age: 24
                    }
                    resolve (`my roll no is ${indexdata} .My name is ${biodata.name} and I am ${biodata.age} years old .`);

                },2000, indexdata)


            });
        }


    //promise consume
    pobj1.then((roll_no) => {    //then method is used to get the result of operation performen using promises.
        console.log(roll_no);
       return  getBiodata(roll_no[1]);
            console.log(kuchbhi);
        }).then((kuchbhi)=>{ console.log(kuchbhi);
        })

    .catch((error)=>{
        console.log(error);     //catch is a method for getting result of error 
    })
</script>